mantidimaging/gui/windows/main/presenter.py

```diff
@@ -563,3 +553,31 @@ def _add_recon_to_dataset(self, recon_data: Images, stack_id: uuid.UUID) -> None
         self.view.create_new_stack(recon_data)
         self.add_recon_item_to_tree_view(parent_id, recon_data.id, len(self.model.datasets[parent_id].recons))
         self.view.model_changed.emit()
+
+    def add_sinograms_to_dataset_and_update_view(self, sino_stack: Images, original_stack_id: uuid.UUID):
+        """
+        Adds sinograms to a dataset or replaces an existing one.
+        :param sino_stack: The sinogram stack.
+        :param original_stack_id: The ID of a stack in the dataset.
+        """
+        parent_id = self.model.get_parent_strict_dataset(original_stack_id)
```

Comment:
Can we make this work with the mixed dataset as well.

Use case would be loading some projections using "File->Load Images" and then creating sinograms. The sinograms would be added to the same MixedDataset.